### PR TITLE
https://github.com/OpenPaaS-Suite/esn-frontend-calendar/issues/51 implement ESNDavImportClient

### LINKED ISSUE #51: Import menu entry does not work
The calendar (+) button -> **Import** does not work.

Error in the console:
```
angular-strap.js:2501 Error: [$injector:unpr] Unknown provider: davImportServiceProvider <- davImportService <- CalCalendarImportController
http://errors.angularjs.org/1.3.20/$injector/unpr?p0=davImportServiceProvider%20%3C-%20davImportService%20%3C-%20CalCalendarImportController
```

### PR DESCRIPTION
**Pull request by tuanlc**
Partial resolve https://github.com/OpenPaaS-Suite/esn-frontend-calendar/issues/51

### CODE REVIEW

File: src/ESNDavImportClient.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { davImportApi, Client } from 'esn-api-client/src';
+
+export default class ESNDavImportClient {
+  constructor(uploadFile) {
+    if (!uploadFile || typeof uploadFile !== 'function') {
```
Comment (itswillta) [inline src/ESNDavImportClient.js]: Isn't this enough already?

```suggestion
    if (typeof uploadFile !== 'function') {
```

Comment (tuanlc) [reply src/ESNDavImportClient.js]: Done

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "no-use-before-define": 0,
+    "consistent-return": 0,
+    "prefer-arrow-callback": 0,
+    "no-unused-expressions": 0,
+    "array-callback-return": 0,
+    "no-param-reassign": 0,
+    "max-len": 0,
+    "indent": 0
+  }
+}
```
Comment (chamerling) [inline .eslintrc.json]: I used this one to match our ESN style:

```json
{
  "extends": "linagora-esn",
  "rules": {
    "consistent-this": ["warn", "self"],
    "no-unused-vars": ["error", { "vars": "all", "args": "after-used" }],
    "no-console": ["error", { "allow": ["warn", "error"] }],
    "no-process-env": "off",
    "arrow-parens": ["error", "as-needed"]
  },
  "parserOptions": {
    "sourceType": "module"
  }
}

```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "esn-dav-import-client",
+  "version": "0.0.1",
+  "description": "JS library for OpenPaaS ESN APIs",
```
Comment (chamerling) [inline package.json]: Wrong description

File: src/ESNDavImportClient.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+export default class ESNDavImportClient {
+  constructor(uploadFile) {
+    if (typeof uploadFile !== 'function') {
+      throw new Error('uploadFile is required and must be a function');
+    }
+
+    this.uploadFile = uploadFile;
+
+    const OPENPAAS_URL = process.env.OPENPAAS_URL || 'http://localhost:8080';
```
Comment (chamerling) [inline src/ESNDavImportClient.js]: Should come from the contructor

**Review by chamerling — CHANGES_REQUESTED** [review]

**Review by chamerling — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

**Pull request merged**